src/software/new_geom/line.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include "software/new_geom/point.h"
+#include "software/new_geom/polynomial.h"
+
+/**
+ * A 2D line.
+ */
+class Line final
+{
+   public:
+    /**
+     * Creates a degenerate Line with zero Points and zero Polynomials
+     */
+    Line();
+
+    /**
+     * Creates a Line from two Points
+     *
+     * @param first the first Point
+     * @param second the second Point
+     */
+    explicit Line(const Point& first, const Point& second);
+
+    /**
+     * Returns the first Point
+     *
+     * @return the first Point
+     */
+    Point getFirst() const;
+
+    /**
+     * Returns the second Point
+     *
+     * @return the second Point
+     */
+    Point getSecond() const;
+
+    /**
+     * Returns the slope
+     *
+     * @return the slope
+     */
+    double getSlope() const;
+
+    /**
+     * Sets the first Point
+     *
+     * @param first the new first Point
+     */
+    void setFirst(const Point& first);
+
+    /**
+     * Sets the second Point
+     *
+     * @param second the new second Point
+     */
+    void setSecond(const Point& second);
+
+    /**
+     * Calculates the Point evaluated at val, starting from the first Point and moving val
+     * units along the Line towards the second Point
```

Comment:
could we change this so 0 is at the first point and 1 is at the second point? Just makes some operations (like evaluating N equally spaced
 points along a line between two points) easier

Reply:
This is a line so it has infinite length so does it even sense to have a first and second point? I'd rather see point and vector

Reply:
Changed internal representation, so discussion is no longer relevant